policy/parser.go

```diff
@@ -32,13 +32,25 @@ const (
 	firstMatch
 )
 
+// NewPolicy creates a policy object which references a policy source and source information.
+func NewPolicy(src *Source, info *ast.SourceInfo) *Policy {
+	return &Policy{
+		metadata: map[string]any{},
+		source:   src,
+		info:     info,
+		semantic: firstMatch,
+	}
+}
+
 // Policy declares a name, rule, and evaluation semantic for a given expression graph.
 type Policy struct {
 	name     ValueString
 	rule     *Rule
 	semantic semanticType
 	info     *ast.SourceInfo
 	source   *Source
+
+	metadata map[string]any
```

Comment:
Are we only adding it to Policy? If we want to parse some rule-specific metadata, it will be a bit harder to store.